### PR TITLE
Baseline completions and signature help

### PR DESCRIPTION
**Pull request by sandersn**
Background: I need to update a lot of jsdoc tests since #41877 makes the jsdoc parse tree much more complete. This leads to a lot of painstaking, repetitive work updating fourslash tests.

In this PR, I add baselining to completions and signature help, to match the baselining for quick info. I also changed the filename, which should really be the containing filename.

## Limitations ## 

- I *only* changed the tests that I care about for #41877 because I wanted to have time to inspect the baselines for this change. And I haven't planned out a way to update the tests. Ideas are welcome.
- Quick info, signature help, smart selection and completions all use the same filename and are thus mutually exclusive in one test. Fourslash tests are mostly divided this way already, so I thought it was an acceptable limit. I had to split five or six existing, gigantic, horrible tests as a result.

### CODE REVIEW

**Review by sheetalkamat — APPROVED** [review]

**Comment by sheetalkamat** [comment]
One suggestion is that the baseline is appended so that you can add multiple commands baseline to same file .. I think it would be useful if you want to edit a file and see completions or quick info after the edit

**Comment by sandersn** [comment]
Or it might be good to add a baselineAll variant that takes a list of baselines to produce.

Another problem I thought of is that the completions baselines are large, so asking for completions at every marker is unneeded.

File: tests/baselines/reference/completionsCommentsClass.baseline [hunk 1]
```diff
@@ -0,0 +1,3755 @@
+[
+  {
+    "marker": {
+      "fileName": "/tests/cases/fourslash/completionsCommentsClass.ts",
+      "position": 599
```
Comment (sheetalkamat) [inline tests/baselines/reference/completionsCommentsClass.baseline:5]: It would be good to have marker name or line and coloumn info as well.

Comment (sandersn) [reply tests/baselines/reference/completionsCommentsClass.baseline:5]: Name is quite easy to add, so I did that. I don't know of an easy way to convert positions back to line/columns, though.

**Pull request merged**